include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -0,0 +1,339 @@
+//===--- DependencyScan.h - C API for Swift Dependency Scanning ---*- C -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This C API is primarily intended to serve as the Swift Driver's
+// dependency scanning facility (https://github.com/apple/swift-driver).
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_C_DEPENDENCY_SCAN_H
+#define SWIFT_C_DEPENDENCY_SCAN_H
+
+#include "DependencyScanMacros.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/// The version constants for the SwiftDependencyScan C API.
+/// SWIFTSCAN_VERSION_MINOR should increase when there are API additions.
+/// SWIFTSCAN_VERSION_MAJOR is intended for "major" source/ABI breaking changes.
+#define SWIFTSCAN_VERSION_MAJOR 0
+#define SWIFTSCAN_VERSION_MINOR 1
+
+SWIFTSCAN_BEGIN_DECLS
+
+//=== Public Scanner Data Types -------------------------------------------===//
+
+/**
+ * A character string used to pass around dependency scan result metadata.
+ * Lifetime of the string is strictly tied to the object whose field it
+ * represents. When the owning object is released, string memory is freed. Use
+ * \c swiftscan_get_C_string() to retrieve the string data.
+ */
+typedef struct {
+  const void *data;
+  unsigned length;
+} swiftscan_string_ref_t;
+
+typedef struct {
+  swiftscan_string_ref_t *strings;
+  unsigned count;
+} swiftscan_string_set_t;
+
+typedef enum {
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_TEXTUAL = 0,
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_BINARY = 1,
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_PLACEHOLDER = 2,
+  SWIFTSCAN_DEPENDENCY_INFO_CLANG = 3
+} swiftscan_dependency_info_kind_t;
+
+/// Opaque container of the details specific to a given module dependency.
+typedef struct swiftscan_module_details_s *swiftscan_module_details_t;
+
+/// Opaque container to a dependency info of a given module.
+typedef struct swiftscan_dependency_info_s *swiftscan_dependency_info_t;
+
+/// Opaque container to an overall result of a dependency scan.
+typedef struct swiftscan_dependency_graph_s *swiftscan_dependency_graph_t;
+
+/// Opaque container to contain the result of a dependency prescan.
+typedef struct swiftscan_import_set_s *swiftscan_import_set_t;
+
+/// Full Dependency Graph (Result)
+typedef struct {
+  int count;
```

Comment:
Nit: maybe use `size_t` for array lengths in ABI?  Also, your other arrays mostly put the pointer first.